src/frequenz/client/reporting/_client.py

```diff
@@ -252,3 +257,82 @@ def dt2ts(dt: datetime) -> PBTimestamp:
         except grpcaio.AioRpcError as e:
             print(f"RPC failed: {e}")
             return
+
+
+# pylint: disable=too-many-arguments
+async def cumulative_energy(
+    client: ReportingApiClient,
```

Comment:
> If it's not part of the client, it needs an URL and key. I guess part of the client makes things simpler now.

No, it just needs a client, which you should have already instantiated. It really exactly the same, is just a syntax issue. The method would be `async def cumulative_energy(self: ReportingApiClient,...)` the same, it is just called `self` by convention. You can actually also call a method like this `ReportingApiClient.cumulative_energy(client, ...)`.

I like utilities being separated because it keeps the client interface clear, otherwise is really hard to draw a line between what should be part of the client what shouldn't.

> @llucax Any thoughts on this? This helper function IMO goes beyond a thin client and could be seen as additional tooling on top of the client. We do think it is overkill to start that now though and wait for further cases.

I think we should adopt the https://github.com/frequenz-floss/frequenz-dispatch-python approach if there is a room for a higher level interface. At least IMHO that worked out well. In *dispatch* we basically have the https://github.com/frequenz-floss/frequenz-client-dispatch-python as a thin wrapper over the API and then `frequenz-dispatch-python` provides a higher level interface, which makes a lot of assumptions about how the API should be used, so it is much less flexible, but much easier to use for 99% of the cases (scientifically proven :P).